src/commands/create-card.js

```diff
@@ -7,7 +10,9 @@ const printOutput = require("../lib/helpers/print-output");
  * @param {import('yargs').Yargs} yargs - Instance of yargs
  */
 const builder = yargs => {
-	return yargs
+	const baseOptions = flow([withToken, withJson]);
```

Comment:
I read up on `flow` but I _may_ not get it. Are these functions that are being called after you pass in the column and PR options?

Reply:
Like what would be the alternative way of doing this - without using lodash?

Reply:
Great question! The alternative would be:

```javascript
return withJson(withToken(yargs))
	.option("column", {
		describe: "Project column ID",
		demandOption: true,
		type: "number"
	})
	.option("pull-request", {
		describe: "Pull request ID",
		demandOption: true,
		type: "number"
	});
```

Right now, this isn't terrible, but as we move options out into these shared helpers it would end up super nested e.g.

```javascript
withAnotherThing(withSomethingElse(withSomething(withJson(withToken(yargs)))))
```

Lodash's `flow` method helps us out like this:

- It accepts an array of functions e.g. `const baseOptions = flow([withToken, withJson]);`
- It returns a function that we can pass an argument into e.g. `baseOptions(yargs)`
- It executes each function serially (one after another), taking the return value from each function and using it as the argument that it passes to the next function

In our case, our `with*` helper methods accept `yargs` as an argument and they also return that same object. `return yargs.option(...)` gives us an instance of `yargs` as `yargs` implements a [fluent interface](https://en.wikipedia.org/wiki/Fluent_interface#JavaScript). You can see that in the code for `yargs`, the `option()` method returns `self`, which is the same object that the `option()` method itself is being defined on (`self.option`): https://github.com/yargs/yargs/blob/51876e69c71e9861fb09847530eeaec9be534f5f/yargs.js#L577-L687

This leads us to...

```
const baseOptions = flow([withToken, withJson]);

// Calling this function that `flow` has returned has the following effect:
// baseOptions(yargs) -> withToken(argumentGivenTo_baseOptions) -> withJson(returnValueFrom_withToken)

const result = baseOptions(yargs);

// `result` is the return value from `withJson`, the last function in the chain of functions that `flow` called. In our case here, that is `yargs`, so we can go and chain other `.option()` calls on to it, yay!
```

Reply:
Good question. I think the answer would involve changing src/lib/helpers/yargs/options.js ... and/or duplicating the common `yargs` options (i.e. `token` and `json`) for each `yargs` command.

Reply:
Is this directly replaceable with the [proposed pipeline operator](https://github.com/tc39/proposal-pipeline-operator)

Which would be 

```js
const result = yargs
	|> baseOptions
	|> withToken
	|> withJson;
```

Reply:
@leggsimon Yep! Bring it on (although the syntax isn't the nicest). In this case it would actually be:

```javascript
const baseOptions = yargs
	|> withToken
	|> withJson;
```

As `baseOptions` was just the identifier for the function returned by Lodash's `flow` method.

Reply:
Yes that makes so much more sense and I totally understand why using `flow` is right for this situation! Thanks so much Simon, this was so friggin' helpful.